components/camel-google-pubsub/src/main/java/org/apache/camel/component/google/pubsub/GooglePubsubEndpoint.java

```diff
@@ -57,16 +58,24 @@
     @UriParam(name = "concurrentConsumers", description = "The number of parallel streams consuming from the subscription", defaultValue = "1")
     private Integer concurrentConsumers = 1;
 
-    @UriParam(name = "maxMessagesPerPoll", description = "The max number of messages to receive from the server in a single API call", defaultValue = "1")
-    private Integer maxMessagesPerPoll = 1;
-
-    @UriParam(name = "connectionFactory", description = "ConnectionFactory to obtain connection to PubSub Service. If non provided the default one will be used")
-    private GooglePubsubConnectionFactory connectionFactory;
-
     @UriParam(defaultValue = "AUTO", enums = "AUTO,NONE", description = "AUTO = exchange gets ack'ed/nack'ed on completion. NONE = downstream process has to ack/nack explicitly")
     private GooglePubsubConstants.AckMode ackMode = GooglePubsubConstants.AckMode.AUTO;
 
-    private Pubsub pubsub;
+    private RemovalListener<String, Publisher> removalListener = removal -> {
+        Publisher publisher = removal.getValue();
+        publisher.shutdown();
```

Comment:
My IDE reports that `publisher` reports this could be `null`, does it make sense to check for nullity here?